app/com/arpnetworking/metrics/portal/query/impl/NoQueryExecutor.java

```diff
@@ -33,8 +34,10 @@
 @Singleton
 public class NoQueryExecutor implements QueryExecutor {
     @Override
-    public CompletionStage<MetricsQueryResult> executeQuery(final MetricsQuery query) throws QueryExecutionException {
```

Comment:
Smaller change but still relevant - I removed the checked exception because IMO it doesn't make sense to both return a completion stage and possibly throw an exception in the main thread. If there are any query errors, they should be thrown in the CompletionStage monad and handled later via an `exceptionally` combinator or `whenComplete`.